### PR TITLE
feat: database extension registry

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Allow the implementation of custom extensions to customize databases via `setupExtensions.ts`.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

WIP

### TESTING INSTRUCTIONS

Writing tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx [hunk 1]
```diff
@@ -437,6 +458,33 @@ const ExtraOptions = ({
...
+      {dbConfigExtensions?.map?.(extension => {
+        const Extension =
+          extension.component as FunctionComponent<IExtensionProps>;
+        return (
+          <Collapse.Panel
+            header={
+              <div>
+                {extension.logo ? (
+                  <img
+                    alt="dbt Cloud"
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx:470]: is this alt meant to be hard coded?

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 2]
```diff
@@ -566,6 +602,93 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
 
   const [useSSHTunneling, setUseSSHTunneling] = useState<boolean>(false);
 
+  const createPostProcessingCallbacks = useRef<any>({});
+  const updatePostProcessingCallbacks = useRef<any>({});
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:606]: If I understand this correctly, it looks like the child is sending a function from a lower level component up to the a higher component by attaching it to a ref, is that right? Would it be easier to import the extension into this parent index component, and then you can call the create and update callbacks later in the parent and then pass down just the visual components to the ExtraOptions? That way you don't have to pass the callbacks up to the parent. I'm not sure if that's necessarily the solution, but I know that these extension points are complicated, so trying to help suggest a simpler solution if there is one.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx (hunk 1 above)
Comment (betodealmeida) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx:470]: Ah, sorry, that needs to go.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx [hunk 3]
```diff
@@ -16,9 +16,13 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React, { ChangeEvent, EventHandler } from 'react';
+import React, { ChangeEvent, EventHandler, FunctionComponent } from 'react';
```
Comment (betodealmeida) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx:19]: This is no longer needed and is causing tests to fail.

**Review by betodealmeida — COMMENTED** [review]
Nice! I like the simplified approach. I can work on the `onsave` method in shell, if you haven't started it.

File: superset-frontend/src/features/databases/DatabaseModal/styles.ts [hunk 4]
```diff
@@ -573,6 +573,7 @@ export const StyledStickyHeader = styled.div`
   top: 0;
   z-index: ${({ theme }) => theme.zIndex.max};
   background: ${({ theme }) => theme.colors.grayscale.light5};
+  height: ${({ theme }) => theme.gridUnit * 16}px;
```
Comment (eschutho) [inline superset-frontend/src/features/databases/DatabaseModal/styles.ts:576]: fixing this height so that it doesn't jump when loading.

**Review by hughhhh — APPROVED** [review]

**Pull request merged**